File: codewars/7kyu/automorphic-number-(special-numbers-series-#6)/dduck-automorphic-number-(special-numbers-series-#6).py
# codewars.com/kata/5a55f04be6be383a50000187/

from unittest import TestCase

automorphic = lambda n: "Automorphic" if n == int(str(n ** 2)[-len(str(n)):]) else "Not!!"
# inspired from JavaScript best solution


TestCase().assertEqual(automorphic(1), "Automorphic")
TestCase().assertEqual(automorphic(3), "Not!!")
TestCase().assertEqual(automorphic(6), "Automorphic")
TestCase().assertEqual(automorphic(9), "Not!!")
TestCase().assertEqual(automorphic(25), "Automorphic")
TestCase().assertEqual(automorphic(53), "Not!!")
TestCase().assertEqual(automorphic(76), "Automorphic")
TestCase().assertEqual(automorphic(95), "Not!!")
TestCase().assertEqual(automorphic(625), "Automorphic")
TestCase().assertEqual(automorphic(225), "Not!!")
print("<COMPLETEDIN::>")

File: codewars/7kyu/automorphic-number-(special-numbers-series-#6)/dduck-automorphic-number-(special-numbers-series-#6).js
// https://www.codewars.com/kata/5a58d889880385c2f40000aa/

const automorphic = (n) => {
  const convertedDigits = Number(String(n ** 2)
    .split('')
    .slice(-String(n).length)
    .join(''));
  if (n === convertedDigits) {
    return 'Automorphic';
  }
  return 'Not!!';
};

console.log(automorphic(1), 'Automorphic');
console.log(automorphic(3), 'Not!!');
console.log(automorphic(6), 'Automorphic');
console.log(automorphic(9), 'Not!!');
console.log(automorphic(25), 'Automorphic');
console.log(automorphic(53), 'Not!!');
console.log(automorphic(76), 'Automorphic');
console.log(automorphic(95), 'Not!!');
console.log(automorphic(625), 'Automorphic');
console.log(automorphic(225), 'Not!!');

/*
best practice in codewars
const automorphic = n => String(n*n).endsWith(String(n)) ? "Automorphic" : "Not!!" ;
 */
